docs/README.md

```diff
@@ -0,0 +1,476 @@
+
+To access a single list, build an `IList<T>` and provide the record key where the list will be stored:
+
+```C#
+ var list = ListBuilder.Configure(clientProvider)
+    .WithDataContext(new DataContext("my_namespace", "my_set"))
+    .UseMessagePackSerializer()
+    .Build<TestType>("list_record_key"); // A bin name can also be specified here
+```
+
+To operate on multiple lists that contain the same data type, build an `IListOperator<T>` by omitting the key when building, i.e:
+
+```C#
+var listOperator = ListBuilder.Configure(clientProvider)
+    .WithDataContext(new DataContext("my_namespace", "my_set"))
+    .UseMessagePackSerializer()
+    .Build<TestType>();
+```
+
+### ReadAll()
+
+Both available interfaces accept a generic type during configuration (and therefore only operate on one type of list) in order to avoid mixing data types in a single list. To illustrate the problem, consider the `ReadAll()` method that returns all items in a list. Without knowing the size of the list, it must deserialize every item into some data type. If an item of a different type is accidentally added to the list, it will be impossible to deserialize without handling the different type specially. The interfaces are designed to (hopefully) prevent this from happening.
+
+If a complex data structure involving lists with multiple types is needed, use the [Operator](#operator) data access object.
+
+## Operator
+
+The `IOperator` interface allows you to build and execute multi-operation transactions on a single record. Generally, this wraps the [`client.Operate`](https://www.aerospike.com/docs/client/csharp/usage/kvs/multiops.html) functionality provided by Aerospike.
+
+`IOperator` currently only supports reading one result from a arbitrarily long list of operations. i.e., you may perform multiple operations at once (e.g. remove an item from a list in one bin, add an item to a list in another bin), but only retrieve one piece of data back (e.g. get the size of the second list).
+
+For example:
+
+```C#
+// Configure an operator
+_operator = OperatorBuilder
+    .Configure(clientProvider)
+    .WithDataContext(new DataContext("my_namespace", "my_set"))
+    .UseMessagePackSerializerWithLz4Compression()
+    .Build();
+    
+var list2Size = await _operator.Key(RecordKey) // The record you're operating on
+    .List.RemoveByValue("list_1_bin", listItem)
+    .List.Append("list_2_bin", listItem)
+    .SizeAsync("list_2_bin", cancellationToken); // Return the result of the "size" operation
+```
+
+### Batch Operations
+
+_This feature is not yet implemented._
+
+Aerospike does not support batching update or write operations on multiple records in the same request. It does, however, support [fetching batches of records](https://docs.aerospike.com/docs/guide/batch.html#known-limitations). The [KeyValueStore](#keyvaluestore) data access object fetches records in batches when reading multiple keys of the same type.
```

Comment:
Should this be removed from the documentation until it is implemented?

Reply:
I think it should be, yeah. Nice catch!